packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -356,56 +364,83 @@ export class TransformerResolver implements TransformerResolverProvider {
-      $util.qr($ctx.stash.put("conditions", []))
-      $util.qr($ctx.stash.put("metadata", {}))
-      $util.qr($ctx.stash.metadata.put("dataSourceType", "${dataSourceType}"))
-      $util.qr($ctx.stash.metadata.put("apiId", "${api.apiId}"))
-      $util.qr($ctx.stash.put("connectionAttributes", {}))
+      ${stashString({ name: 'typeName', value: this.typeName })}
+      ${stashString({ name: 'fieldName', value: this.fieldName })}
+      ${stashExpression({ name: 'conditions', value: '[]' })}
+      ${stashExpression({ name: 'metadata', value: '{}' })}
+      ${stashString({ name: 'dataSourceType', value: dataSourceType, object: 'metadata' })}
+      ${stashString({ name: 'apiId', value: api.apiId, object: 'metadata' })}
+      ${stashExpression({ name: 'connectionAttributes', value: '{}' })}
       ${dataSource}
     `;
     const account = Stack.of(context.stackManager.scope).account;
     const authRole = context.synthParameters.authenticatedUserRoleName;
     if (authRole) {
+      const authRoleArn = `arn:aws:sts::${account}:assumed-role/${authRole}/CognitoIdentityCredentials`;
+      const authRoleStatement = stashString({ name: 'authRole', value: authRoleArn });
+
       initResolver += dedent`\n
-      $util.qr($ctx.stash.put("authRole", "arn:aws:sts::${account}:assumed-role/${authRole}/CognitoIdentityCredentials"))
+        ${authRoleStatement}
       `;
     }
     const unauthRole = context.synthParameters.unauthenticatedUserRoleName;
     if (unauthRole) {
+      const unauthRoleArn = `arn:aws:sts::${account}:assumed-role/${unauthRole}/CognitoIdentityCredentials`;
+      const unauthRoleStatement = stashString({ name: 'unauthRole', value: unauthRoleArn });
       initResolver += dedent`\n
-      $util.qr($ctx.stash.put("unauthRole", "arn:aws:sts::${account}:assumed-role/${unauthRole}/CognitoIdentityCredentials"))
+        ${unauthRoleStatement}
       `;
     }
     const identityPoolId = context.synthParameters.identityPoolId;
     if (identityPoolId) {
+      const identityPoolStatement = stashString({ name: 'identityPoolId', value: identityPoolId });
       initResolver += dedent`\n
-        $util.qr($ctx.stash.put("identityPoolId", "${identityPoolId}"))
+        ${identityPoolStatement}
       `;
     }
     const adminRoles = context.synthParameters.adminRoles ?? [];
+    const adminRolesStatement = stashExpression({ name: 'adminRoles', value: JSON.stringify(adminRoles) });
     initResolver += dedent`\n
-      $util.qr($ctx.stash.put("adminRoles", ${JSON.stringify(adminRoles)}))
+      ${adminRolesStatement}
     `;
-    initResolver += '\n$util.toJson({})';
+
+    if (isJsResolverFnRuntime(this.runtime)) {
+      initResolver = dedent`
+        export const request = (ctx) => {
+          ${initResolver}
+          return {};
+        }
+      `;
+    } else {
+      initResolver += '\n$util.toJson({})';
+    }
+
+    const initResponseResolver = isJsResolverFnRuntime(this.runtime)
```

Comment:
I'm not a fan of introducing more such conditionals in existing packages based on runtime. Something to think about long term is to have runtime specific generators defined. That would consolidate all the runtime specific generations/overrides.

Reply:
Totally agree. The alternative meant a much larger (and imo riskier) code change that should be done, but done in isolation.